pkg/template/resource.go

```diff
@@ -46,40 +47,61 @@ type getClusterTriggerBinding func(name string, options metav1.GetOptions) (*tri
 // ResolveTrigger takes in a trigger containing object refs to bindings and
 // templates and resolves them to their underlying values.
 func ResolveTrigger(trigger triggersv1.EventListenerTrigger, getTB getTriggerBinding, getCTB getClusterTriggerBinding, getTT getTriggerTemplate) (ResolvedTrigger, error) {
-	tb := make([]*triggersv1.TriggerBinding, 0, len(trigger.Bindings))
-	ctb := make([]*triggersv1.ClusterTriggerBinding, 0, len(trigger.Bindings))
-	for _, b := range trigger.Bindings {
-		if b.Spec != nil {
-			tb = append(tb, &triggersv1.TriggerBinding{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: b.Name,
-				},
-				Spec: *b.Spec,
+	bp, err := ResolveBindingsToParams(trigger.Bindings, getTB, getCTB)
+	if err != nil {
+		return ResolvedTrigger{}, fmt.Errorf("failed to resolve bindings: %w", err)
+	}
+	if trigger.Template.Name == "" {
+		return ResolvedTrigger{}, fmt.Errorf("trigger template name is empty")
+	}
```

Comment:
I'm 50/50 on this - should we remove this? This should already be validated to ensure the name is there (https://github.com/tektoncd/triggers/blob/f73c5f12a03f063e448d414f96e1b18e2ed2185e/pkg/apis/triggers/v1alpha1/trigger_validation.go#L66), and if it is empty it should cause the getTT to fail

Reply:
yeah that makes sense. removed.